dev/planning-example.py

```diff
@@ -0,0 +1,206 @@
+obs.pressure = 0.600 * u.bar
+obs.relative_humidity = 0.2
+obs.temperature = 10 * u.deg_C
+
+# Dates are assumed to be UTC by default
+
+"""
+We can ask about typical times of interest at this observing position
+Returned dates are assumed to be in the observer's time zone for convenience.
+The date parameter, if given, can be an astropy Time or a datetime instance.
+If no parameter is given, the current date/time is assumed.
+"""
+
+# Define a time
+from astropy.time import Time
+time_obs = Time(2457189.500000, format='jd')
+
+# sunset
+obs.sunset(time=time_obs, which='nearest') # Default
+obs.sunset(time=time_obs, which='next')
+obs.sunset(time=time_obs, which='previous')
+obs.sunset() # Will be use time=Time.now(), which='nearest'
+
+# sunrise
+obs.sunrise(time=time_obs, which='nearest') # Default
+obs.sunrise(time=time_obs, which='next')
+obs.sunrise(time=time_obs, which='previous')
+
+# moon rise
+obs.moonrise(time=time_obs, which='nearest')
+
+# moon set
+obs.moonset(time=time_obs, which='nearest')
+
+# The above functions can be called with an `angle` keyword argument to specify
+# a particular horizon angle for rising or setting, or can be called with
+# convenience functions for particular morning/evening twilight.
+# For example, to compute astronomical twilight by specifying the `angle`:
+obs.sunset(time=time_obs, which='next', angle=18*u.degree) 
+
+# evening (astronomical) twilight
+obs.evening_astronomical(time=time_obs)
+
+# evening (nautical) twilight
+obs.evening_nautical(time=time_obs)
+
+# evening (civil) twilight
+obs.evening_civil(time=time_obs)
+
+# morning (nautical) twilight
+obs.morning_nautical(time=time_obs)
+
+# morning (civil) twilight
+obs.morning_civil(time=time_obs)
+
+# morning (astronomical) twilight
+obs.morning_astronomical(time=time_obs)
+
+# what is the moon illumination?
+# returns a float, which is percentage of the moon illuminated
+obs.moon_illumination(time=time_obs)
+
+# what is the moon altitude and azimuth?
+obs.moon_altaz(time=time_obs)
+
+# Other sun-related convenience functions:
+obs.noon(time=time_obs, which='nearest')
+obs.midnight(time=time_obs, which='nearest')
+
+# ==============
+# Target objects
+# ==============
+'''
+A target object defines a single observation target, with coordinates
+and an optional name.
+'''
+from astroplan import FixedTarget, airmass_plot
+
+# Define a target.
+
+from astropy.coordinates import SkyCoord
+t1 = FixedTarget(name='Polaris',
+                 coord=SkyCoord('02h31m49.09s', '+89d15m50.8s', frame='icrs'))
+
+# Leaves scope for NonSiderealTarget, etc.
+
+# Convenience methods for looking up/constructing targets by name via
+# astroquery:
+t1 = FixedTarget.from_name('Polaris')
+
+# ================================
+# Condition objects, observability
+# ================================
+
+"""
+Q: Can I observe this target on the night of May 1, 2015 between 18:30
+and 05:30 HST, above airmass 1.2, on a telescope that can observe between 
```

Comment:
Please bear with me ... I'm a gamma-ray astronomer that never used airmass. I know what altitude and zenith angle is.

What does "above airmass 1.2" mean?
I presume usually the constraint people want is to observe at low zenith angles = high altitudes, right?
And that corresponds to low airmass, so usually people want an upper limit as an airmass constraint, no?
https://en.wikipedia.org/wiki/Air_mass_(astronomy)#/media/File:AirmassFormulaePlots.png

But in this API spec and at http://astroplan.readthedocs.org/en/latest/api/astroplan.AboveAirmass.html we only allow to specify a lower limit.

Here's what I think:
* The relation of airmass and altitude should be explained in the airmass constraint docstring
* The user should be able to constrain airmass low / high, just like it's possible for altitude. The default could be `None` for both, meaning no constraint on either min or max?
* I would prefer something regular for min / max constraints ... use `TimeRange`, `AltitudeRange` and `AirmassRange`. (at the moment we have [AboveAirmass](http://astroplan.readthedocs.org/en/latest/api/astroplan.AboveAirmass.html), [AltitudeRange](http://astroplan.readthedocs.org/en/latest/api/astroplan.AltitudeRange.html), [TimeWindow](http://astroplan.readthedocs.org/en/latest/api/astroplan.TimeWindow.html).